### PR TITLE
Jenkins backend fix "no last build"

### LINKED ISSUE #6725: Jenkins pluging ci/cd tab is not displaying build information
## Expected Behavior

Would like to see all build information in the CI/CD tab.

## Current Behavior

I see the Jenkins summary build (master) in the overview page. However ci/cd tab is not able to display other builds. See below exception in the logs. 

backstage error Cannot read property 'building' of null type=errorHandler stack=TypeError: Cannot read property 'building' of null
    at _JenkinsApiImpl.augmentProject (/app/node_modules/@backstage/plugin-jenkins-backend/dist/index.cjs.js:65:34)
    at _JenkinsApiImpl.getProjects (/app/node_modules/@backstage/plugin-jenkins-backend/dist/index.cjs.js:34:30)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async /app/node_modules/@backstage/plugin-jenkins-backend/dist/index.cjs.js:196:22

## Your Environment

- NodeJS Version (v12):
- Operating System and Version (e.g. Ubuntu 14.04):
- Browser Information: Chrome

### PR DESCRIPTION
**Pull request by GregoireW**
## Hey, I just made a Pull Request!

This PR fix #6725.  Jenkins may have jobs without lastBuild, so backstage need to manage correctly that.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/jenkins-backend/src/service/jenkinsApi.ts [hunk 1]
```diff
@@ -165,7 +168,10 @@ export class JenkinsApiImpl {
 
     return {
       ...project,
-      lastBuild: this.augmentBuild(project.lastBuild, jobScmInfo),
+      lastBuild:
+        project.lastBuild === null
```
Comment (freben) [inline plugins/jenkins-backend/src/service/jenkinsApi.ts:172]: Would like something like

```ts
project.lastBuild
  ? this.augmentBuild(project.lastBuild, jobScmInfo)
  : undefined;
```

Preferring to treat "as a boolean", and avoiding null :)

**Review by freben — COMMENTED** [review]
Thanks, awesome! Just a nit on null / boolean

File: plugins/jenkins-backend/src/service/jenkinsApi.ts (hunk 1 above)
Comment (GregoireW) [reply plugins/jenkins-backend/src/service/jenkinsApi.ts]: I can set the  `project.lastBuild?....`, but I cannot set "undefined" as jenkins send null and the same object is used as base for both the jenkins mapping and backstage object. 
Just did the change.

File: plugins/jenkins-backend/src/types.ts [hunk 2]
```diff
@@ -63,7 +63,7 @@ export interface BackstageBuild extends CommonBuild {
 
 export interface CommonProject {
   // standard Jenkins
-  lastBuild: CommonBuild;
+  lastBuild: CommonBuild | null;
```
Comment (Rugvip) [inline plugins/jenkins-backend/src/types.ts:66]: I'm also thinking that we should avoid the use of null in our representation of these, so these and the other ones would instead be `lastBuild?: CommonBuild`. If you feel it's needed we could create separate types for the API return types and then convert, but tbh that seems a bit overkill.

Comment (GregoireW) [reply plugins/jenkins-backend/src/types.ts:66]: Jenkins return  "lastBuild: null".    If I put "lastBuild?: CommonBuild" I cannot test this part as typescript won't let me compile the code.

**Review by freben — APPROVED** [review]
Yeah I'm ok with this as a quick fix, separating the models would be nice but could be done separately. In its current state it essentially fixes the bug by not missing the null in the declared return type, which feels good on its own

**Pull request merged**